app/models/concerns/csv_exporter.rb

```diff
@@ -0,0 +1,17 @@
+require 'csv'
+
+module CsvExporter
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def to_csv(columns: column_names, csv_options: {})
+      CSV.generate(csv_options) do |csv|
+        csv << columns.map(&:to_s)
+        columns.map!(&:to_sym)
+        all.limit(nil).find_each do |obj|
```

Comment:
`find_each` loads them in batches of 1000 by default.
Also, this method is called on a scope that includes the search filter.